### ISSUE TITLE
Add "get_server_commands" to Arkouda client

### ISSUE THREAD

**Issue by glitch** [Bears-R-Us/arkouda#1049]
Now that we have a modular build process where commands can be optionally added at compilation time, we need a way to query the server for the list/map of available commands.  

Since Arkouda is currently caching the `ServerConfig` I think it makes sense to capture the CommandMap contents in its own message procedure at this time.  The separation would also keep the json generation the message responses a bit cleaner.

The expected output would be a python dictionary (map) of `command_name` -> `function_name`, which would be a straight dump of the `CommandMap.commandMap` contents.  (Note: since this is determined at compile time, I think it's ok to cache/memoize this as a string on the server on startup.)

Additionally if we wanted to add a `list_server_modules` command we could maintain a list of the Modules which were built via `ServerModules.cfg` plus the required/"backbone" modules included in `MsgProcessing.chpl`.

To recap, new client commands

- `get_server_commands` -> yields dictionary of `{"command name" : "function name"}' (dump of the command map)
- Optionally, `list_server_modules` -> yields list of `["arkouda module name"]`

**Comment by glitch** [comment]
Based on what we already have, I think the Msg portion likely goes in `MsgProcessing.chpl` based upon what is already in there (i..e `getconfigMsg`, `getmemusedMsg`, `infoMsg`).  Also given the command map is now an integral part of the system/server I think it makes sense that it is not an optional part of the build process, so it makes sense to live in this core module.

**Comment by glitch** [comment]
Couple of things I stumbled across while implementing this. Just dumping the contents of CommandMap isn't enough because we have a number of "required" commands which do not live in the command map for a few reasons.
- Function args don't match
- Function return val doesn't match

For example cmd "array" -> arrayMsg takes 4 args instead of the standard 3.
For example cmd "tondarray" -> tondarrayMsg returns bytes instead of MsgTuple

I think this gives us some room for improvement.  I'm going to add to the CommandMap a `registerBinaryFunction` and store binary funcs in their own map.  I'm also going to restructure the cmd parsing section in `arkouda_server` so that these required functions get add to the CommandMap but anything that requires special handling (i.e. `connect`, `shutdown`, etc.) gets processed first before we check the command map.  I'm going to put dummy function signatures into the map for these special case functions as well so they naturally show up as a list of commands.
